pkg/callbackserver/callbackserver.go

```diff
@@ -100,7 +94,16 @@ func (cs *CallbackServer) Run(ctx context.Context, callback CallbackFunc) (inter
 		break
 	}
 
-	return result, microerror.Mask(origErr)
+	err := server.Shutdown(ctx)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	if origErr != nil {
```

Comment:
Added this explicit error check, to not rely on `microerror.Mask()`'s underlying implementation